File: HANDS-ON/07.Week-7/Day-32/1. Contact Management Web Application (Using Dependency Injection)/Services/ContactService.cs
using WebApplication5.Models;

namespace WebApplication5.Services
{
    public class ContactService : IContactService
    {
        private static List<ContactInfo> contacts = new List<ContactInfo>();

        public List<ContactInfo> GetAllContacts()
        {
            return contacts;
        }
        public ContactInfo GetContactById(int id)
        {
            return contacts.FirstOrDefault(c => c.ContactId == id);
        }

       public void AddContact(ContactInfo contact)
        {
            contacts.Add(contact);
        }
    }
}

File: HANDS-ON/11.Week-11/Day-52/1.Contact_Management_HttpClient/app/contact.service.ts
import { Injectable } from '@angular/core';
import { Contact } from '../Models/Contact';
import { HttpClient } from '@angular/common/http';

@Injectable({
  providedIn: 'root'
})
export class ContactService {

    readonly API_URL:string = "https://localhost:7110/api/Contacts/";
  
    constructor(private httpClient:HttpClient){}
  
    public getContacts(){
      return this.httpClient.get<Contact[]>(this.API_URL);
    }

    addContact(contact:Contact){
   
    return this.httpClient.post<Contact>(this.API_URL,contact);
 }

  getContactById(id:number){
  return this.httpClient.get<Contact>(this.API_URL+id);
 }

 updateContact(contact:Contact){
  return this.httpClient.put(this.API_URL+contact.id,contact);
 }

 deleteContact(id:number){
  return this.httpClient.delete(this.API_URL+id);
 }

  
}

 

  

